### ISSUE TITLE
Bug: ...

### ISSUE THREAD

**Issue by HaythamB** [qdm12/ddns-updater#115]
**TLDR**: config.json can't parse properly for Dreamhost updates; failing to bind the key value from the json config. file Same error applies when I add the config to the docker-compose variable too.

1. Is this urgent?

    - [ ] Yes
    - [x] No

2. What DNS service provider(s) are you using?

    - [] Cloudflare
    - [ ] DDNSS.de
    - [ ] DonDominio
    - [ ] DNSOMatic
    - [ ] DNSPod
    - [x] Dreamhost
    - [ ] DuckDNS
    - [ ] DynDNS
    - [ ] GoDaddy
    - [ ] Google
    - [ ] He.net
    - [ ] Infomaniak
    - [ ] Namecheap
    - [ ] NoIP

3. What's the version of the program?
Running version latest built on 2020-10-31T14:41:32Z (commit 3240bb7)

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```
2020-11-11T20:23:37.516Z	INFO	Found single setting to update record
2020-11-11T20:23:37.804Z	INFO	Reading history from database: domain example.com host @
2020-11-11T20:23:37.805Z	INFO	Web UI listening at address 0.0.0.0:8000 with root URL ""
2020-11-11T20:23:37.809Z	INFO	backup: disabled
2020-11-11T20:23:38.227Z	INFO	IPv4 address of example.com is 86.195.72.192 and your IPv4 address is 86.195.72.196
2020-11-11T20:23:38.228Z	INFO	Updating record [domain: example.com | host: @ | provider: dreamhost | ip: ipv4] to use 86.195.72.196
panic: assignment to entry in nil map
goroutine 53 [running]:
net/url.Values.Set(...)
	net/url/url.go:898
github.com/qdm12/ddns-updater/internal/settings.makeDreamhostDefaultValues(0xfbab50, 0x10, 0xcb0040)
	github.com/qdm12/ddns-updater/internal/settings/dreamhost.go:142 +0x6c
github.com/qdm12/ddns-updater/internal/settings.listDreamhostRecords(0x71df30, 0x104a220, 0x71d510, 0x104a100, 0xfbab50, 0x10, 0x0, 0x0, 0xfaf880, 0x65, ...)
	github.com/qdm12/ddns-updater/internal/settings/dreamhost.go:154 +0x68
github.com/qdm12/ddns-updater/internal/settings.(*dreamhost).Update(0x10c9920, 0x71df30, 0x104a220, 0x71d510, 0x104a100, 0xfbbe00, 0x10, 0x10, 0x8, 0x0, ...)
	github.com/qdm12/ddns-updater/internal/settings/dreamhost.go:100 +0x84
github.com/qdm12/ddns-updater/internal/update.(*updater).Update(0x104a140, 0x71df30, 0x104a220, 0x0, 0xfbbe00, 0x10, 0x10, 0x8d99d3c3, 0xbfe32ff2, 0x3677fe97, ...)
	github.com/qdm12/ddns-updater/internal/update/update.go:46 +0x114
github.com/qdm12/ddns-updater/internal/update.(*runner).updateNecessary(0xd001b0, 0x71df30, 0x104a220)
	github.com/qdm12/ddns-updater/internal/update/run.go:218 +0x4a0
github.com/qdm12/ddns-updater/internal/update.(*runner).Run.func1(0xec4180, 0xd001b0, 0x71df30, 0x104a220, 0xc4a080)
	github.com/qdm12/ddns-updater/internal/update/run.go:233 +0xc8
created by github.com/qdm12/ddns-updater/internal/update.(*runner).Run
	github.com/qdm12/ddns-updater/internal/update/run.go:227 +0x80
```

Configuration file (**remove your credentials!**):

```
{ "settings": [
{    "provider": "dreamhost",
    "domain": "XXXXX",
    "host": "@",
    "key": "XXXXXXXXXXXXXXXX",
    "ip_version": "ipv4"
  }]
}
```

Host OS:
Raspbian

**Comment by HaythamB** [comment]
Oh and another factor: you have to add the "host" key:value in the config, or it will also throw an error! This should be a separate issue I know! :D

**Comment by qdm12** [comment]
Whoa thanks for reporting the issue. Definitely some bad coding from my side for Dreamhost. I'll fix it in a few hours!

**Comment by HaythamB** [comment]
Thanks. I've narrowed it down to the Dreamhost constructor function for the struct. Seems like its passing the entire JSON object, rather than the individual settings object from within the array ;)

**Comment by qdm12** [comment]
Done, thanks for the help!

Let me know if anything doesn't work. Sorry, I never tested Dreamhost thoroughly 😕 

And yes, that bug forcing you to specify a host was silly too!

**Comment by HaythamB** [comment]
Works perfectly! Thanks for the super fast turnaround time with that! 👍
